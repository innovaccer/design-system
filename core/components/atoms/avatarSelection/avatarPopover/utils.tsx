import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef?.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    const searchInput = listRef.current.querySelector('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput.focus();
    setFocusedOption && setFocusedOption(searchInput);
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
