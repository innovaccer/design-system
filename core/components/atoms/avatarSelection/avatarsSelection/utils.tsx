import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput && searchInput[0]?.focus();
    setFocusedOption?.(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
