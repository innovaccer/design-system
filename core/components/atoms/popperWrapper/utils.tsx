// Helper function to convert CSS class names to style objects
export const createStyleFromClass = (className: string) => {
  if (!className) return {};

  // Create a temporary element to get computed styles
  const tempElement = document.createElement('div');
  tempElement.className = className;
  document.body.appendChild(tempElement);

  const computedStyle = window.getComputedStyle(tempElement);
  const style: React.CSSProperties = {};

  // Extract relevant animation properties
  const animation = computedStyle.animation;
  const animationDuration = computedStyle.animationDuration;
  const animationTimingFunction = computedStyle.animationTimingFunction;
  const animationFillMode = computedStyle.animationFillMode;
  const opacity = computedStyle.opacity;
  const transform = computedStyle.transform;

  if (animation && animation !== 'none') {
    style.animation = animation;
  }
  if (animationDuration && animationDuration !== '0s') {
    style.animationDuration = animationDuration;
  }
  if (animationTimingFunction && animationTimingFunction !== 'ease') {
    style.animationTimingFunction = animationTimingFunction;
  }
  if (animationFillMode && animationFillMode !== 'none') {
    style.animationFillMode = animationFillMode;
  }
  if (opacity && opacity !== '1') {
    style.opacity = opacity;
  }
  if (transform && transform !== 'none') {
    style.transform = transform;
  }

  document.body.removeChild(tempElement);
  return style;
};
