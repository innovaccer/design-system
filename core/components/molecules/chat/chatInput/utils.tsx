import { MessageType } from './ChatInput';

export const positionMentionPopup = (
  textareaRef: React.RefObject<HTMLDivElement>,
  setMentionPosition: React.Dispatch<React.SetStateAction<{ top: number; left: number }>>
) => {
  if (!textareaRef.current) return;

  const selection = window.getSelection();
  if (!selection || !selection.rangeCount) return;

  const range = selection.getRangeAt(0);
  const rect = range.getBoundingClientRect();
  const editorRect = textareaRef.current.getBoundingClientRect();

  // Calculate position based on the cursor position
  let top = rect.bottom + window.scrollY;
  let left = rect.left + window.scrollX;

  // Adjust position if it would go off-screen
  if (top + 150 > window.innerHeight) {
    top = rect.top + window.scrollY - 150;
  }
  if (left + 200 > window.innerWidth) {
    left = window.innerWidth - 200;
  }

  // Adjust position relative to the editor
  top -= editorRect.top + window.scrollY - 12;
  left -= editorRect.left + window.scrollX - 8;

  setMentionPosition({ top, left });
};

export const extractMessageData = (textareaRef: React.RefObject<HTMLDivElement>): MessageType[] => {
  const elements = textareaRef.current?.childNodes;

  return Array.from(elements || []).map((item: any) => {
    if (item?.dataset?.type === 'mention') {
      const mentionValue = JSON.parse(item.dataset.content);
      return { type: 'mention', content: mentionValue?.data, nodeName: item.nodeName };
    }

    // handles new line by pressing enter
    else if (item.innerText === '\n' && item.nodeName === 'DIV') {
      return { type: 'newline', content: '\n', nodeName: item.nodeName };
    }

    // handles line break by pressing shift + enter
    else if (item.nodeName === 'BR') {
      return { type: 'linebreak', content: '<br>', nodeName: item.nodeName };
    }

    return { type: 'text', content: item.innerText || item.textContent, nodeName: item.nodeName };
  });
};
