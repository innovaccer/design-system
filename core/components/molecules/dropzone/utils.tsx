const isIe = (userAgent: string) => {
  return userAgent.indexOf('MSIE') !== -1 || userAgent.indexOf('Trident/') !== -1;
};

const isEdge = (userAgent: string) => {
  return userAgent.indexOf('Edge/') !== -1;
};

export const accepts = (file: File, acceptedFiles?: string | string[]) => {
  if (file && acceptedFiles) {
    const acceptedFilesArray = Array.isArray(acceptedFiles) ? acceptedFiles : acceptedFiles.split(',');
    const fileName = file.name || '';
    const mimeType = (file.type || '').toLowerCase();
    const baseMimeType = mimeType.replace(/\/.*$/, '');

    if (!mimeType) {
      return true;
    }

    return acceptedFilesArray.some((type) => {
      const validType = type.trim().toLowerCase();
      if (validType.charAt(0) === '.') {
        return fileName.toLowerCase().endsWith(validType);
      }
      if (validType.endsWith('/*')) {
        return baseMimeType === validType.replace(/\/.*$/, '');
      }
      return mimeType === validType;
    });
  }
  return true;
};

export const isPropagationStopped = (event: any) => {
  if (typeof event.isPropagationStopped === 'function') {
    return event.isPropagationStopped();
  }
  if (typeof event.cancelBubble !== 'undefined') {
    return event.cancelBubble;
  }
  return false;
};

export const isEvtWithFiles = (event: any) => {
  if (!event.dataTransfer) {
    return !!event.target && !!event.target.files;
  }

  return Array.prototype.some.call(
    event.dataTransfer.types,
    (type) => type === 'Files' || type === 'application/x-moz-file'
  );
};

export const isKindFile = (item: null | Record<string, unknown>) => {
  return typeof item === 'object' && item !== null && item.kind === 'file';
};

export const onDocumentDragOver = (event: DragEvent) => {
  event.preventDefault();
};

export const isIeOrEdge = (userAgent = window.navigator.userAgent) => {
  return isIe(userAgent) || isEdge(userAgent);
};

type functype = (event: Event, ...args: unknown[]) => void;

export const composeEventHandlers = (...fns: functype[]) => {
  return (event: Event, ...args: unknown[]) =>
    fns.some((fn) => {
      if (!isPropagationStopped(event) && fn) {
        fn(event, ...args);
      }
      return isPropagationStopped(event);
    });
};

export const reducer = (state: any, action: any) => {
  switch (action.type) {
    case 'focus':
      return {
        ...state,
        isFocused: true,
      };
    case 'blur':
      return {
        ...state,
        isFocused: false,
      };
    case 'openDialog':
      return {
        ...state,
        isFileDialogActive: true,
      };
    case 'closeDialog':
      return {
        ...state,
        isFileDialogActive: false,
      };
    case 'setDraggedFiles':
      /* eslint no-case-declarations: 0 */
      const { isDragActive, draggedFiles } = action;
      return {
        ...state,
        draggedFiles,
        isDragActive,
      };
    case 'setFiles':
      return {
        ...state,
        acceptedFiles: action.acceptedFiles,
        fileRejections: action.fileRejections,
      };
    case 'reset':
      return {
        ...state,
        isFileDialogActive: false,
        isDragActive: false,
        draggedFiles: [],
        acceptedFiles: [],
        fileRejections: [],
      };
    default:
      return state;
  }
};
