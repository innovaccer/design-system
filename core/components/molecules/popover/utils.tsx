import { Placement } from '@floating-ui/react';
import { PositionType as Position } from '@/common.type';

export const mapPositionToPlacement = (position: Position): Placement => {
  const positionMap: Record<Position, Placement> = {
    auto: 'bottom',
    'auto-start': 'bottom-start',
    'auto-end': 'bottom-end',
    'top-start': 'top-start',
    top: 'top',
    'top-end': 'top-end',
    'right-start': 'right-start',
    right: 'right',
    'right-end': 'right-end',
    'bottom-end': 'bottom-end',
    bottom: 'bottom',
    'bottom-start': 'bottom-start',
    'left-end': 'left-end',
    left: 'left',
    'left-start': 'left-start',
  };

  return positionMap[position] || 'bottom';
};
