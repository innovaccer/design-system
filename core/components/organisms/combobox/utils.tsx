import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  inputTriggerRef?: any,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  multiSelect?: boolean,
  listRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef);
      break;
    case 'Enter':
      handleEnterKey(focusedOption, multiSelect, inputTriggerRef, listRef, setFocusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      inputTriggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (
  focusedOption: Element | undefined,
  multiSelect?: boolean,
  inputTriggerRef?: any,
  listRef?: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>
) => {
  (focusedOption as HTMLElement)?.click();
  if (!multiSelect) {
    inputTriggerRef.current.focus();
  } else {
    // to focus first option by default when last option is selected
    const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
    const listArr = Array.from(listItems);
    const index = listArr.findIndex((item) => {
      return item == focusedOption;
    });

    if (index === listArr.length - 1) {
      (listItems[0] as HTMLElement).focus();
      setFocusedOption && setFocusedOption(listItems[0]);
      listItems[0].scrollIntoView({ block: 'center' });
    }
  }
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any
) => {
  const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;
  }

  const targetOption = listItems[index];

  (targetOption as HTMLElement).focus();
  setFocusedOption && setFocusedOption(targetOption);
  targetOption?.scrollIntoView?.({ block: 'center' });
};
