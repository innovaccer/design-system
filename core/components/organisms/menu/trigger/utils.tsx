import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    case 'ArrowDown':
      event.preventDefault();
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'Escape':
    case 'Tab':
      setOpenPopover?.(false);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();

  if (targetOption && typeof targetOption.scrollIntoView === 'function') {
    (targetOption as HTMLElement)?.scrollIntoView({ block: 'end' });
  }
  setFocusedOption && setFocusedOption(targetOption);
};
