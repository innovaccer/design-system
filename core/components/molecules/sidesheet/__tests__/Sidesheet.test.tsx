import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { SidesheetProps as Props } from '@/index.type';
import { Button, Sidesheet, Text } from '@/index';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const dimension = ['regular', 'large'];
const FunctionValue = jest.fn();
const seperator = false;
const backIcon = false;
const stickFooter = false;
const backdropClose = false;
const footer = (
  <>
    <Button appearance="primary" className="mr-4" data-test="DesignSystem-Sidesheet--SidesheetButton">
      Primary
    </Button>
    <Button appearance="basic">Basic</Button>
  </>
);
const header = <Text data-test="DesignSystem-Sideheet--SidesheetText">Heading</Text>;

const headerOptions = {
  backIcon,
  heading: 'Heading',
  subHeading: 'Subheading',
};

const mapper = {
  open: valueHelper(true, { required: true }),
  backdropClose: valueHelper(backdropClose, { required: true }),
  dimension: valueHelper(dimension, { required: true, iterate: true }),
  headerOptions: valueHelper(headerOptions, { required: true }),
  seperator: valueHelper(seperator, { required: true }),
  stickFooter: valueHelper(stickFooter, { required: true }),
};

describe('Sidesheet component', () => {
  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Sidesheet {...attr} />);

      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('sidesheet component', () => {
  it('renders children with props: headerOptions and footerOptions', () => {
    const { getByTestId } = render(
      <Sidesheet dimension="large" headerOptions={headerOptions} open={true} footerOptions={{ actions: [] }} />
    );

    expect(getByTestId('DesignSystem-SidesheetContainer')).toBeInTheDocument();
    expect(getByTestId('DesignSystem-Sidesheet')).toBeInTheDocument();
    expect(getByTestId('DesignSystem-Sidesheet--Header')).toBeInTheDocument();
    expect(getByTestId('DesignSystem-Sidesheet--CloseButton')).toBeInTheDocument();
    expect(getByTestId('DesignSystem-Sidesheet')).toHaveClass('Sidesheet--open');
  });

  it('renders children with props: header and footer', () => {
    const { getByTestId } = render(<Sidesheet dimension="large" header={header} open={true} footer={footer} />);

    expect(getByTestId('DesignSystem-SidesheetContainer')).toBeInTheDocument();
    expect(getByTestId('DesignSystem-Sidesheet')).toBeInTheDocument();
    expect(getByTestId('DesignSystem-Sidesheet--Header')).toBeInTheDocument();
    expect(getByTestId('DesignSystem-Sidesheet--CloseButton')).toBeInTheDocument();
    expect(getByTestId('DesignSystem-Sidesheet--Footer')).toBeInTheDocument();
    expect(getByTestId('DesignSystem-Sidesheet--SidesheetButton')).toBeInTheDocument();
    expect(getByTestId('DesignSystem-Sideheet--SidesheetText')).toBeInTheDocument();
    expect(getByTestId('DesignSystem-Sidesheet')).toHaveClass('Sidesheet--open');
  });

  it('renders children without footer and footerOptions', () => {
    const { queryByTestId } = render(<Sidesheet dimension="large" header={header} open={true} stickFooter={false} />);
    expect(queryByTestId('DesignSystem-Sidesheet--Footer')).not.toBeInTheDocument();
    expect(queryByTestId('DesignSystem-Sidesheet--OverlayBody')).not.toHaveClass('Sidesheet-body--withMargin');
    expect(
      queryByTestId('DesignSystem-Sidesheet--Footer')?.classList.contains('Sidesheet-footer--stickToBottom')
    ).toBeFalsy();
  });

  it('renders children with footer sticked to bottom', () => {
    const { getByTestId } = render(
      <Sidesheet dimension="large" header={header} open={true} footer={footer} stickFooter={true} />
    );
    expect(getByTestId('DesignSystem-Sidesheet--Footer')).toHaveClass('Sidesheet-footer--stickToBottom');
    expect(getByTestId('DesignSystem-Sidesheet--OverlayBody')).toHaveClass('Sidesheet-body--withMargin');
  });

  it('renders children with seperator', () => {
    const { getByTestId } = render(
      <Sidesheet dimension="large" header={header} open={true} footer={footer} seperator={true} />
    );
    expect(getByTestId('DesignSystem-Sidesheet--Footer')).toHaveClass('Sidesheet-footer--withSeperator');
    expect(getByTestId('DesignSystem-Sidesheet--Header')?.parentNode).toHaveClass('Sidesheet-header--withSeperator');
  });

  it('calls onClose function with backdropClose being true', () => {
    const { getByTestId } = render(
      <>
        <div data-test="DesignSystem-OutsideClick">Outside Click</div>
        <Sidesheet
          backdropClose={true}
          dimension="large"
          headerOptions={headerOptions}
          open={true}
          footer={footer}
          onClose={FunctionValue}
        >
          <Text>Modal Body</Text>
        </Sidesheet>
      </>
    );

    const OutsideClick = getByTestId('DesignSystem-OutsideClick');
    fireEvent.click(OutsideClick);
    expect(FunctionValue).toHaveBeenCalled();
  });
});

describe('Sidesheet component with prop: dimension', () => {
  it('renders Sidesheet with dimension: large', () => {
    const { getByTestId } = render(
      <Sidesheet dimension="large" headerOptions={headerOptions} open={true} footer={footer} />
    );
    expect(getByTestId('DesignSystem-Sidesheet')).toHaveClass('Col--10');
  });

  it('renders Sidesheet with dimension: regular', () => {
    const { getByTestId } = render(
      <Sidesheet dimension="regular" headerOptions={headerOptions} open={true} footer={footer} />
    );
    expect(getByTestId('DesignSystem-Sidesheet')).toHaveClass('Col--6');
  });
});

describe('Sidesheet component with prop: open', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.runOnlyPendingTimers();
    jest.useRealTimers();
  });

  it('renders Sidesheet with open: true', () => {
    const { getByTestId } = render(
      <Sidesheet dimension="large" headerOptions={headerOptions} open={true} footer={footer} />
    );

    expect(getByTestId('DesignSystem-Sidesheet')).toHaveClass('Sidesheet--open');
    expect(getByTestId('DesignSystem-Sidesheet')).toHaveClass('Sidesheet-animation--open');
    expect(getByTestId('DesignSystem-SidesheetContainer')).toHaveClass('Overlay-container--open');
  });

  it('renders Sidesheet with open: false', () => {
    const { getByTestId } = render(
      <Sidesheet dimension="large" headerOptions={headerOptions} open={false} footer={footer} />
    );

    expect(getByTestId('DesignSystem-Sidesheet')).toHaveClass('Sidesheet-animation--close');
  });

  it('renders Sidesheet with toggle of open', () => {
    const { getByTestId, rerender } = render(
      <Sidesheet
        backdropClose={backdropClose}
        dimension="large"
        headerOptions={headerOptions}
        seperator={seperator}
        open={true}
        footer={footer}
      >
        <Text>Modal Body</Text>
      </Sidesheet>
    );

    expect(getByTestId('DesignSystem-Sidesheet')).toHaveClass('Sidesheet--open');
    expect(getByTestId('DesignSystem-Sidesheet')).toHaveClass('Sidesheet-animation--open');

    const closeIcon = getByTestId('DesignSystem-Sidesheet--CloseButton');
    fireEvent.click(closeIcon);

    rerender(
      <Sidesheet
        backdropClose={backdropClose}
        dimension="large"
        headerOptions={headerOptions}
        seperator={seperator}
        open={false}
        footer={footer}
      >
        <Text>Modal Body</Text>
      </Sidesheet>
    );

    expect(getByTestId('DesignSystem-Sidesheet')).toHaveClass('Sidesheet-animation--close');

    jest.runOnlyPendingTimers();
    jest.useRealTimers();
  });
});
