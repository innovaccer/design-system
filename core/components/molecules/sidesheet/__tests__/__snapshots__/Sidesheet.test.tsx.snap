// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidesheet component 
  open: true, backdropClose: false, dimension: "large", headerOptions: {"backIcon":false,"heading":"Heading","subHeading":"Subheading"}, seperator: false, stickFooter: false
 1`] = `
<body
  style="overflow: hidden;"
>
  <div
    class="Overlay-wrapper"
  >
    <div
      class="Row Overlay-container fade-in Overlay-container--open"
      data-layer="true"
      data-open="true"
      data-test="DesignSystem-SidesheetContainer"
      style="z-index: 1001;"
    >
      <div
        class="Col Col--10 Sidesheet Sidesheet--open Sidesheet-animation--open"
        data-test="DesignSystem-Sidesheet"
      >
        <div
          class="Sidesheet-header"
        >
          <div
            class="Col"
            data-test="DesignSystem-Sidesheet--Header"
          >
            <div
              class="OverlayHeader"
              data-test="DesignSystem-OverlayHeader"
            >
              <div
                class="OverlayHeader-headerWrapper"
              >
                <h4
                  class="Heading Heading--m Heading--default Sidesheet-header--shiftLeft"
                  data-test="DesignSystem-OverlayHeader--heading"
                >
                  Heading
                </h4>
              </div>
              <span
                class="Text Text--subtle Text--regular OverlayHeader-subheading"
                data-test="DesignSystem-OverlayHeader--Subheading"
              >
                Subheading
              </span>
            </div>
          </div>
          <div
            class="Col flex-grow-0"
            data-test="DesignSystem-Column"
          >
            <style />
            <div
              class="PopperWrapper-trigger"
            >
              <button
                class="Button Button--regular Button--regularSquare Button--transparent"
                data-test="DesignSystem-Sidesheet--CloseButton"
                tabindex="0"
              >
                <div
                  class="Button-icon"
                  data-test="DesignSystem-Button--Icon-Wrapper"
                >
                  <i
                    class="material-symbols material-symbols-rounded Icon"
                    data-test="DesignSystem-Button--Icon"
                    role="button"
                    style="font-size: 20px; width: 20px;"
                  >
                    close
                  </i>
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          class="OverlayBody Sidesheet-body Sidesheet-body--firstPage"
          data-test="DesignSystem-Sidesheet--OverlayBody"
        />
      </div>
    </div>
  </div>
  <div />
  <div
    class="Backdrop Backdrop--open Backdrop-animation--open"
    data-layer="true"
    data-opened="true"
    data-test="DesignSystem-Backdrop"
    style="z-index: 1000;"
  />
</body>
`;

exports[`Sidesheet component 
  open: true, backdropClose: false, dimension: "regular", headerOptions: {"backIcon":false,"heading":"Heading","subHeading":"Subheading"}, seperator: false, stickFooter: false
 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    class="Overlay-wrapper"
  >
    <div
      class="Row Overlay-container fade-in Overlay-container--open"
      data-layer="true"
      data-open="true"
      data-test="DesignSystem-SidesheetContainer"
      style="z-index: 1001;"
    >
      <div
        class="Col Col--6 Sidesheet Sidesheet--open Sidesheet-animation--open"
        data-test="DesignSystem-Sidesheet"
      >
        <div
          class="Sidesheet-header"
        >
          <div
            class="Col"
            data-test="DesignSystem-Sidesheet--Header"
          >
            <div
              class="OverlayHeader"
              data-test="DesignSystem-OverlayHeader"
            >
              <div
                class="OverlayHeader-headerWrapper"
              >
                <h4
                  class="Heading Heading--m Heading--default Sidesheet-header--shiftLeft"
                  data-test="DesignSystem-OverlayHeader--heading"
                >
                  Heading
                </h4>
              </div>
              <span
                class="Text Text--subtle Text--regular OverlayHeader-subheading"
                data-test="DesignSystem-OverlayHeader--Subheading"
              >
                Subheading
              </span>
            </div>
          </div>
          <div
            class="Col flex-grow-0"
            data-test="DesignSystem-Column"
          >
            <style />
            <div
              class="PopperWrapper-trigger"
            >
              <button
                class="Button Button--regular Button--regularSquare Button--transparent"
                data-test="DesignSystem-Sidesheet--CloseButton"
                tabindex="0"
              >
                <div
                  class="Button-icon"
                  data-test="DesignSystem-Button--Icon-Wrapper"
                >
                  <i
                    class="material-symbols material-symbols-rounded Icon"
                    data-test="DesignSystem-Button--Icon"
                    role="button"
                    style="font-size: 20px; width: 20px;"
                  >
                    close
                  </i>
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          class="OverlayBody Sidesheet-body Sidesheet-body--firstPage"
          data-test="DesignSystem-Sidesheet--OverlayBody"
        />
      </div>
    </div>
  </div>
  <div
    class="Backdrop Backdrop--open Backdrop-animation--open"
    data-layer="true"
    data-opened="true"
    data-test="DesignSystem-Backdrop"
    style="z-index: 1000;"
  />
</body>
`;
